### DISCUSSION TITLE
Within a Prisma $transaction, can you use the output of one create() as the input of another?

### DISCUSSION THREAD

**Discussion by miller-productions** [prisma/prisma#4641] [Q&A] [answered]
Referring to this example here on the Prisma website...

https://www.prisma.io/docs/concepts/components/prisma-client/transactions#the-transaction-api

```ts
const write1 = prisma.user.create()
const write2 = prisma.post.create()
const write3 = prisma.profile.create()
const [result1, result2, result3] = await prisma.$transaction([write1, write2, write3])
```

Would it be possible to use the output from `write1` as the input inside the create() function for `write2`.

Another way to ask the question, let's say you want to run a transaction in prisma where you first create one record, returning say an ID, and then you use that ID as an input to another create that happens in the same prisma transaction?

I suspect it might be possible within `write2` by referencing `(await write1).ID` for example?

What do you think?

---
For example, assuming the schema looks like this...

```prisma
model User {
  id           Int       @id @default(autoincrement())
  name         String?
  email        String    @unique
  posts        Post[]
}

model Post {
  id         Int     @id @default(autoincrement())
  title      String
  author     User    @relation(fields:  [authorId], references: [id])
  authorId   Int
}
```

Could you possibly do something like this...

```ts
const write1 = prisma.user.create({
    data: {
        email: "some@emailaddress.com",
    },
});

const write2 = prisma.post.create({
    data: {
        title: "some title",
        authorId: (await write1).id,
    },
});

const write3 = prisma.profile.create(); // not really relevant to this question

const [result1, result2, result3] = await prisma.$transaction([write1, write2, write3]);

**Comment by ryands17** [comment, marked as answer]
Hey @miller-productions 👋 

Prisma's transactions are different than traditional transactions, which is why it's not possible to infer the previous operation value currently in the next one. You can read more about how Prisma supports transactions [here](https://www.prisma.io/blog/how-prisma-supports-transactions-x45s1d5l0ww1).

In the above model, it would be better to use [nested-writes](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries/#update-an-existing-user-record-by-connect-it-to-two-existing-post-records-or-creating-two-new-post-records) instead.

So your query would look like this:

```ts
const singleWrite = prisma.post.create({
    data: {
      title: 'some title',
      author: { create: { email: 'email: "some@emailaddress.com",' } },
    },
  })
```

Let me know if your use-case is different here :)

**Reply by miller-productions** [reply]
Hi @ryands17,

Thanks for the explanation and link back to info about transactions.

Nested writes seem like the logical solution. 

The use case I am exploring is fairly complex. I'll close off this discussion and might open a separate one if I can't figure it out.

**Reply by ryands17** [reply]
Awesome 🚀

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
